### PR TITLE
[ASCII-2461] Fix flaky TestRunProviders

### PR DESCRIPTION
**Pull request by pgimalac**
### What does this PR do?
Update `TestRunProviders` to fix a race condition. 
Still check that providers are started, and that we don't wait for them forever.

### Motivation
Not have a race.

### CODE REVIEW

File: comp/core/flare/flare_test.go [hunk 1]
```diff
@@ -130,9 +125,10 @@ func TestRunProviders(t *testing.T) {
 	fb, err := helpers.NewFlareBuilder(false, flarebuilder.FlareArgs{})
 	require.NoError(t, err)
 
+	start := time.Now()
 	f.Comp.(*flare).runProviders(fb, cliProviderTimeout)
+	<-firstStarted
```
Comment (dustmop) [inline comp/core/flare/flare_test.go:130]: One of the things the test appeared to be testing before this PR was that the 2nd flareFiller (with the `time.Sleep(10)`) did not complete; it should be killed by the timeout. Having the assertion that less than 5 seconds have passed is ensuring the same thing, but it's not at all obvious from just reading the test that this is intended.

Is it possible to keep `secondDone` around, and validate that it stays false at the end of the test, while still getting rid of the existing race condition?

Comment (pgimalac) [reply comp/core/flare/flare_test.go:130]: Hum I considered it as pretty much the same thing but fair

Comment (dustmop) [reply comp/core/flare/flare_test.go:130]: Adding some comments explaining the intention would also help. Overall, I agree that the channel is a better way to do the first part (ensuring that the first provider runs).

**Review by dustmop — APPROVED** [review]

File: comp/core/flare/flare_test.go (hunk 1 above)
Comment (pgimalac) [reply comp/core/flare/flare_test.go:133]: I added a couple comments !

**Pull request merged**